In [1]:
def bayes_posterior(prior, likelihood, evidence):
    posterior = (prior * likelihood) / evidence
    return posterior


# Example values
prior = 0.30
likelihood = 0.80
evidence = 0.50

result = bayes_posterior(prior, likelihood, evidence)

print("Posterior probability:", result)
print("Percentage:", result * 100, "%")

Posterior probability: 0.48
Percentage: 48.0 %


Bayes theorem is used to calculate the probability of an event based on previously known information.

Formula:

P(A|B) = (P(B|A) × P(A)) / P(B)

Here:
- Prior = 0.30
- Likelihood = 0.80
- Evidence = 0.50
- Posterior = 0.48

Therefore, the posterior probability is **48%**.

In [2]:
# Given probabilities
premium = 0.10
regular = 0.90

discount_given_premium = 0.80
discount_given_regular = 0.20

# Probability of getting a discount
discount = (discount_given_premium * premium) + (discount_given_regular * regular)

# Bayes theorem
probability_premium = (discount_given_premium * premium) / discount

print("Probability that the restaurant is premium:", probability_premium)
print("Percentage:", probability_premium * 100, "%")

Probability that the restaurant is premium: 0.30769230769230776
Percentage: 30.769230769230777 %


Given:

- P(Premium) = 0.10
- P(Regular) = 0.90
- P(Discount | Premium) = 0.80
- P(Discount | Regular) = 0.20

First, we calculate the probability of getting a discount:

P(Discount) = (0.80 × 0.10) + (0.20 × 0.90)
             = 0.26

Using Bayes theorem:

P(Premium | Discount) = (0.80 × 0.10) / 0.26
                       = 0.3077

Therefore, if a restaurant offers a discount, there is approximately a **30.77% probability that it is a premium restaurant**.

In [4]:
# Given probabilities

# Probability that an email is spam
p_spam = 0.30

# Probability that an email is not spam
p_not_spam = 0.70

# Probability of finding "free" in a spam email
p_free_given_spam = 0.80

# Probability of finding "free" in a non-spam email
p_free_given_not_spam = 0.10

# Calculate probability of "free" appearing in any email
p_free = (p_free_given_spam * p_spam) + (p_free_given_not_spam * p_not_spam)

# Apply Bayes theorem
p_spam_given_free = (p_free_given_spam * p_spam) / p_free

print("Probability that email is spam if it contains 'free':", p_spam_given_free)
print("Percentage:", round(p_spam_given_free * 100, 2), "%")

Probability that email is spam if it contains 'free': 0.7741935483870968
Percentage: 77.42 %


Suppose 30% of emails are spam.

The word "free" appears in 80% of spam emails and 10% of non-spam emails.

Using Bayes theorem, we calculate the probability that an email is spam when it contains the word "free".

The result is approximately **77.42%**.

Therefore, if an email contains the word "free", there is approximately a **77.42% probability that it is spam**.

In [5]:
# Given probabilities

# Probability of buying a mobile phone
p_phone = 0.40

# Probability of not buying a mobile phone
p_not_phone = 0.60

# Probability of buying earphones given that user bought a phone
p_earphones_given_phone = 0.30

# Probability of buying earphones given that user did not buy a phone
p_earphones_given_not_phone = 0.10

# Probability of buying earphones
p_earphones = (
    p_earphones_given_phone * p_phone
    + p_earphones_given_not_phone * p_not_phone
)

# Bayes theorem
p_phone_given_earphones = (
    p_earphones_given_phone * p_phone
) / p_earphones

print("Probability that user bought a phone:", round(p_phone_given_earphones, 2))
print("Percentage:", round(p_phone_given_earphones * 100, 2), "%")

Probability that user bought a phone: 0.67
Percentage: 66.67 %


Suppose 40% of users buy a mobile phone. Among users who buy a mobile phone, 30% also buy earphones. Among users who do not buy a mobile phone, 10% buy earphones.

Using Bayes theorem, the probability that a user bought a mobile phone given that they bought earphones is approximately **66.67%**.

Flipkart could use this information to recommend earphones to users who purchase a mobile phone.